content/articles/creating-a-shopping-cart-api/index.md

```diff
@@ -0,0 +1,562 @@
+        BigDecimal total=BigDecimal.ZERO;
+        for(Product product:productCart){
+       total= total.add(product.getCost());
+        }
+   return total;
+    }
+    public List<Product> displayAllItems(){
+       return productCart;
+    }
+    public void setCartToNull(){
+        productCart.clear();
+    }
+}
+```
+### Controller Package
+The last package to review is the controller package. In the controller package we have our endpoints or APIs that connect to the internet all the basic functionalities of our services:
+```java
+@RestController
+@RequestMapping("cart")
+public class Controller {
+    @Autowired
+    DataServiceImpl dataService;
+    @Autowired
+    CartServiceImpl cartService;
+
+    @PostMapping("/create_customer")
+    public Customer registerCustomer(@RequestBody Customer customer) {
+        return dataService.registerCustomer(customer);
+    }
+
+    @GetMapping("/get_all_customers")
+    public List<Customer> gettingAllList() {
+        return dataService.gettingAllCustomers();
+    }
+
+    @PostMapping("/create_admin")
+    public Admin registerAdmin(@RequestBody Admin admin) {
+        return dataService.registerAdmin(admin);
+    }
+
+    @GetMapping("/get_all_admins")
+    public List<Admin> getAllAdmin() {
+        return dataService.getAllAdmin();
+    }
+
+    @PostMapping("/{id}/add_product")
+    public void addProducts(@PathVariable Long id, @RequestBody Product product) {
+        cartService.addProductToCartByCustomer(id, product);
+    }
+
+    @PutMapping("/{id}/add_product")
+    public void addProduct(@PathVariable Long id, @RequestBody Product product) {
+        cartService.addProductToCartByCustomer(id, product);
+    }
+
+    @GetMapping("/cal_price")
+    public BigDecimal calculatingPrice() {
+        return cartService.calculatePrice();
+    }
+
+    @GetMapping("/{id}/payment")
+    public BigDecimal payment(@PathVariable Long id) {
+        return cartService.calculatingPurchase(id);
+    }
+
+    @GetMapping("/{id}/view_all_product")
+    public List<Product> viewAllProducts(@PathVariable Long id) {
+        return cartService.checkAllTheProductsInTheCartByCustomer(id);
+    }
+
+    @GetMapping("/{id}/view_all_productAdmin")
+    public List<Product> viewAllProduct(@PathVariable Long id) {
+        return cartService.checkAllTheProductsInTheCartByAdmin(id);
+    }
+
+    @DeleteMapping("/{id}/delete_product")
+    public void delete(@PathVariable Long id, Product product) {
+        cartService.removeProductsFromCartByCustomer(id, product);
+    }
+}
+```
+N/B `@PostMapping` is for saving objects, `@GetMapping` is for retrieving data from the database, `@PatchMapping` and `@PutMapping` is for updating data, and `@DeleteMapping` is for deleting. Also, note that we annotate the fields of this class with `@Autowire` to inject its dependencies to avoid initializing new objects. This is done by using these same objects each time they're needed throughout the whole codebase.
```

Comment:
I don't believe you made it clear to a reader which endpoints each of the methods map to and how you are doing it. I would add a comment of brief explanation about how you map each of these methods to a particular URL using the `@Postmapping`, `@GetMapping`, `@PatchMapping`, `@PutMapping`, and `@DeleteMapping` annotations.